lib/Jsrt/ChakraCore.h

```diff
@@ -163,4 +163,394 @@ JsGetModuleHostInfo(
+///     Called by the runtime to load the source code of the serialized script.
+///     The caller must keep the script buffer valid until the JsSerializedScriptUnloadCallback.
+///     This callback is only supported by the Win32 version of the API
+/// </summary>
+/// <param name="sourceContext">The context passed to Js[Parse|Run]SerializedScriptWithCallback</param>
+/// <param name="script">The script returned.</param>
+/// <returns>
+///     true if the operation succeeded, false otherwise.
+/// </returns>
+typedef bool (CHAKRA_CALLBACK * JsSerializedLoadScriptCallback)
+    (JsSourceContext sourceContext, _Out_ JsValueRef *value,
+    _Out_ JsParseScriptAttributes *parseAttributes);
+
+/// <summary>
+///     Create JavascriptString variable from C string
+/// </summary>
+/// <remarks>
+///     <para>
+///         C string is expected to be ASCII
+///     </para>
+/// </remarks>
+/// <param name="content">Pointer to string memory.</param>
+/// <param name="length">Number of bytes within the string</param>
+/// <param name="value">JsValueRef representing the JavascriptString</param>
+/// <returns>
+///     The code <c>JsNoError</c> if the operation succeeded, a failure code otherwise.
+/// </returns>
+CHAKRA_API
+    JsCreateString(
+        _In_ const char *content,
+        _In_ size_t length,
+        _Out_ JsValueRef *value);
+
+/// <summary>
+///     Create JavascriptString variable from Utf8 string
+/// </summary>
+/// <remarks>
+///     <para>
+///         Input string can be either ASCII or Utf8
+///     </para>
+/// </remarks>
+/// <param name="content">Pointer to string memory.</param>
+/// <param name="length">Number of bytes within the string</param>
+/// <param name="value">JsValueRef representing the JavascriptString</param>
+/// <returns>
+///     The code <c>JsNoError</c> if the operation succeeded, a failure code otherwise.
+/// </returns>
+CHAKRA_API
+    JsCreateStringUtf8(
+        _In_ const uint8_t *content,
+        _In_ size_t length,
+        _Out_ JsValueRef *value);
+
+/// <summary>
+///     Create JavascriptString variable from Utf16 string
+/// </summary>
+/// <remarks>
+///     <para>
+///         Expects Utf16 string
+///     </para>
+/// </remarks>
+/// <param name="content">Pointer to string memory.</param>
+/// <param name="length">Number of bytes within the string</param>
+/// <param name="value">JsValueRef representing the JavascriptString</param>
+/// <returns>
+///     The code <c>JsNoError</c> if the operation succeeded, a failure code otherwise.
+/// </returns>
+CHAKRA_API
+    JsCreateStringUtf16(
+        _In_ const uint16_t *content,
+        _In_ size_t length,
+        _Out_ JsValueRef *value);
+
+/// <summary>
+///     Write JavascriptString value into C string buffer
+/// </summary>
+/// <remarks>
+///     <para>
+///         When size of the `buffer` is unknown,
+///         `buffer` argument can be nullptr.
+///         `written` argument will return the size needed.
+///     </para>
```

Comment:
Documentation of JsWriteString/JsWriteStringUtf16 are incorrect (do not match code). These 2 do not return "needed buffer size".